From 27605e21e4dd0b6ed3ed4b2ec5ae92af22822b63 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 5 Feb 2013 18:34:43 +0530
Subject: [PATCH 05/10] ResourceWatcher: Emit the current signals in
 propertyChanged

We were accidentally emitting the wrong paramaters for propertyRemoved
and propertyAdded.
---
 libnepomukcore/datamanagement/resourcewatcher.cpp | 16 ++++++++--------
 libnepomukcore/datamanagement/resourcewatcher.h   |  4 +---
 2 files changed, 9 insertions(+), 11 deletions(-)

diff --git a/libnepomukcore/datamanagement/resourcewatcher.cpp b/libnepomukcore/datamanagement/resourcewatcher.cpp
index 4a741f2..f394ae8 100644
--- a/libnepomukcore/datamanagement/resourcewatcher.cpp
+++ b/libnepomukcore/datamanagement/resourcewatcher.cpp
@@ -318,20 +318,20 @@ namespace {
     }
 }
 
-void Nepomuk2::ResourceWatcher::slotPropertyChanged(const QString& resUri, const QString& propUri, const QVariantList& oldObjs, const QVariantList& newObjs)
+void Nepomuk2::ResourceWatcher::slotPropertyChanged(const QString& res_, const QString& prop_, const QVariantList& addedObjs, const QVariantList& removedObjs)
 {
-    const Resource res = Resource::fromResourceUri(KUrl(resUri));
-    const Types::Property prop = KUrl(propUri);
+    const Resource res = Resource::fromResourceUri(KUrl(res_));
+    const Types::Property prop = KUrl(prop_);
 
-    foreach( const QVariant& v, oldObjs ) {
-        emit propertyRemoved( res, prop, v );
+    foreach( const QVariant& v, addedObjs ) {
+        emit propertyAdded( res, prop, convertType(prop, v) );
     }
 
-    foreach( const QVariant& v, newObjs ) {
-        emit propertyAdded( res, prop, v );
+    foreach( const QVariant& v, removedObjs ) {
+        emit propertyRemoved( res, prop, convertType(prop, v) );
     }
 
-    emit propertyChanged( res, prop, oldObjs, newObjs );
+    emit propertyChanged( res, prop, addedObjs, removedObjs );
 }
 
 #include "resourcewatcher.moc"
diff --git a/libnepomukcore/datamanagement/resourcewatcher.h b/libnepomukcore/datamanagement/resourcewatcher.h
index 621fbf6..05e3710 100644
--- a/libnepomukcore/datamanagement/resourcewatcher.h
+++ b/libnepomukcore/datamanagement/resourcewatcher.h
@@ -352,9 +352,7 @@ namespace Nepomuk2 {
         void slotResourceRemoved(const QString& res, const QStringList& types);
         void slotResourceTypesAdded(const QString& res, const QStringList& types);
         void slotResourceTypesRemoved(const QString& res, const QStringList& types);
-        void slotPropertyChanged(const QString& res, const QString& prop,
-                                 const QVariantList & oldObjs,
-                                 const QVariantList & newObjs);
+        void slotPropertyChanged(const QString& res, const QString& prop_, const QVariantList& addedObjs, const QVariantList& removedObjs);
     private:
         class Private;
         Private * d;
-- 
1.8.1.2

